src/resource_clients/request_queue.ts

```diff
@@ -103,33 +108,65 @@ export class RequestQueueClient extends ResourceClient {
+        ow(requests, ow.array.ofType(ow.object.partialShape({
+            id: ow.undefined,
+        })).minLength(1).maxLength(MAX_REQUESTS_PER_BATCH_OPERATION));
+        ow(options, ow.object.exactShape({
+            forefront: ow.optional.boolean,
+        }));
+
+        const { data } = await this.httpClient.call({
+            url: this._url('requests/batch'),
+            method: 'POST',
+            timeout: this.timeoutMillis,
+            data: requests,
+            params: this._params({
+                forefront: options.forefront,
+                clientKey: this.clientKey,
+            }),
+        });
+
+        return cast(parseDateFields(pluckData(data)));
+    }
+
     protected async _batchAddRequestsWithRetries(
         requests: Omit<RequestQueueClientRequestSchema, 'id'>[],
-        options: RequestQueueClientBatchAddRequestOptions = {},
-    ): Promise<RequestQueueClientBatchAddRequestsResult> {
+        options: RequestQueueClientBatchAddRequestWithRetriesOptions = {},
+    ): Promise<RequestQueueClientBatchRequestsOperationResult> {
+        const {
+            maxUnprocessedRequestsRetries = this.httpClient.maxRetries,
+            forefront,
+        } = options;
         // Keep track of the requests that remain to be processed (in parameter format)
         let remainingRequests = requests;
         // Keep track of the requests that have been processed (in api format)
         const processedRequests: ProcessedRequest[] = [];
         // The requests we have not been able to process in the last call
         // ie. those we have not been able to process at all
         let unprocessedRequests: UnprocessedRequest[] = [];
-        // TODO: Do not use `maxRetries` from http client, but rather a separate value
-        for (let i = 0; i < 1 + this.httpClient.maxRetries; i++) {
+        for (let attempt = 0; attempt < 1 + maxUnprocessedRequestsRetries; attempt++) {
             try {
-                const response = await this.httpClient.call({
-                    url: this._url('requests/batch'),
-                    method: 'POST',
-                    timeout: this.timeoutMillis,
-                    data: remainingRequests,
-                    params: this._params({
-                        forefront: options.forefront,
-                        clientKey: this.clientKey,
-                    }),
+                const response = await this._batchAddRequests(remainingRequests, {
+                    forefront,
                 });
-                const data = response.data.data as RequestQueueClientBatchAddRequestsResult;
-                processedRequests.push(...data.processedRequests);
-                unprocessedRequests = data.unprocessedRequests;
+                processedRequests.push(...response.processedRequests);
+                unprocessedRequests = response.unprocessedRequests;
+
+                // Consider request with unprocessed requests as rate limited.
```

Comment:
Just add note that this is important to let AutoscalingPool in SDK know that it should potentially downscale.